filesource-common/src/main/java/io/cdap/plugin/common/azurecred/AzureClientSecretService.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.common.azurecred;
+
+
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+import io.netty.util.internal.StringUtil;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.alias.CredentialProvider;
+import org.apache.hadoop.security.alias.CredentialProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+
+public class AzureClientSecretService {
+
+  private static Logger logger = LoggerFactory.getLogger(AzureClientSecretService.class);
+
+  /* To read ADLS credentials stored in KeyVault, we need to first access KeyVault Store itself
+   * 1. Either user has provided KeyVault credentials itself in properties : `fs.adl.oauth2.client.id` and `fs.adl.oauth2.credential`
+   * 2. If not, these credentials are encrypted in a hadoop's jceks file.
+   * 3. jceks file path is either specified by user in property - hadoop.security.credential.provider.path
+   * 4. If not, we should get this property value from core-site.xml
+   * 5. If path not provided in core-site.xml also, we use the default jceks path on an azure vm
+   */
+  public static HashMap<String, String> getADLSSecretsUsingJceksAndKV(String keyVaultUrl, HashMap<String, String> credMap,
+                                                                      Map<String, String> properties) {
+    Map<String, String> keymap = properties;
+
+    if (!(keymap.containsKey("fs.adl.oauth2.client.id") && keymap.containsKey("fs.adl.oauth2.credential"))) {
+      // First get the credentials to access KeyVault from Jceks file
+      try {
+        keymap = getSecretsFromJcek("fs.adl.oauth2.client.id,fs.adl.oauth2.credential",
+                                            properties.getOrDefault("hadoop.security.credential.provider.path", getJceksPath()));
+        logger.info("KeyVault access identified");
+      } catch (Exception e) {
+        logger.error("jcek Exception : " + e);
+      }
+    }
+    return getSecretsFromKV(keyVaultUrl, credMap, keymap.get("fs.adl.oauth2.client.id"), keymap.get("fs.adl.oauth2.credential"));
+  }
+
+
+  public static String getJceksPath() {
+
+      String jceks = "jceks://hdfs@mycluster/etc/security/jceks/adls.jceks";  // init with default for Azure
+
+      try {
+          Configuration conf = new Configuration();
+          // Assumption: this path is always correct in Azure for hdfs-site.xml and core-site.xml
```

Comment:
this is not a valid assumption for Hadoop clusters in general. Is this assuming CDAP is running on Azure?